### PR TITLE
Fixed wrong Publish condition

### PR DESCRIPTION
**Pull request by segraef**
# Change

Fixed Publish condition to run only on main and master refs.

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: .github/workflows/ms.analysisservices.servers.yml [hunk 1]
```diff
@@ -117,7 +117,7 @@ jobs:
   ###############
   job_publish_module:
     name: 'Publish module'
-    if: contains(fromJson('["refs/heads/main", "refs/heads/master"]'), github.ref)
+    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/master' || github.event.inputs.prerelease == 'true'
```
Comment (MrMCake) [inline .github/workflows/ms.analysisservices.servers.yml:120]: Beware, the flag does not yet exist in main. I'd propose to merge this without the prerelease flag to main

Comment (MrMCake) [reply .github/workflows/ms.analysisservices.servers.yml:120]: The prerelease condition can then be incorporated in @MariusStorhaug s PR

Comment (segraef) [reply .github/workflows/ms.analysisservices.servers.yml]: Adjusted for this PR, left in #744.

Comment (segraef) [reply .github/workflows/ms.analysisservices.servers.yml]: Ready for approval @MrMCake

Comment (segraef) [reply .github/workflows/ms.analysisservices.servers.yml]: Adjusted for this PR, left in #744. 
Ready for approval @MrMCake.

**Comment by MrMCake** [comment]
@segraef  Merge once no pipelines are running as it will trigger all - and there is no batch processing feature in GitHub

**Review by MrMCake — APPROVED** [review]

**Pull request merged**